tests/conftest.py

```diff
@@ -119,11 +118,119 @@ def generate_tests(self, smoke_test_data, random_test_data=[]):
+    ):
+        """Generate Data that checks for log and exp are inverses. Specifically
+
+            :math: `Exp_{base_point}(Log_{base_point}(point)) = point`
+
+
+        Parameters
+        ----------
+        space : cls
+            Manifold class that upon which metric is present.
+        max_n : int
+            Range of 'n' to generated.
+            Optional, default: 20
+        num_n : int
+            Number of 'n' to be generated.
+            Optional, default: 5
+        num_samples : int
+            Optional, default: 100
+
+        Returns
+        -------
+        _ : list
+            Test Data.
+        """
+        random_n = random.sample(range(1, max_n), num_n)
+        random_data = []
+        for n in random_n:
+            for prod in itertools.product(*kwargs.values()):
+                space_n = space(n)
+                base_point = space_n.random_point(num_samples)
+                point = space_n.random_point(num_samples)
+                random_data.append((n,) + prod + (point, base_point))
+        return self.generate_tests([], random_data)
+
+    def _geodesic_belongs_data(
+        self, space, max_n=10, n_n=5, n_geodesics=10, n_t=10, **kwargs
+    ):
+        """Generate Data that checks for points on geodesic belongs to data. Specifically
+
+        Parameters
+        ----------
+        space : cls
+            Manifold class that upon which metric is present.
+        max_n : int
+            Range of 'n' to generated.
+            Optional, default: 10
+        n_n : int
+            Number of 'n' to be generated.
+            Optional, default: 5
+        n_geodesics : int
+            Number of geodesics to be generated.
+            Optional, default: 10
+        n_t : int
+            Number of points to be sampled on each geodesic.
+            Optional, default: 10
+        Returns
+        -------
+        _ : list
+            Test Data.
+        """
+        random_n = random.sample(range(2, max_n), n_n)
+        random_data = []
+        for n in random_n:
+            for prod in itertools.product(*kwargs.values()):
+                space_n = space(n)
+                initial_point = space_n.random_point()
+                initial_tangent_points = space_n.random_tangent_vec(
+                    n_geodesics, base_point=initial_point
+                )
+                random_t = gs.linspace(start=-1.0, stop=1.0, num=n_t)
+                # for (initial_point, initial_tangent_point), t in itertools.product(
+                #     zip(initial_points, initial_tangent_points), random_t
+                # ):
+                for initial_tangent_point, t in itertools.product(
+                    initial_tangent_points, random_t
+                ):
+                    random_data.append(
+                        (n,) + prod + (initial_point, initial_tangent_point, t)
+                    )
+        return self.generate_tests([], random_data)
+
+    def _squared_dist_is_symmetric_data(
```

Comment:
I know this is a private function, but we still need docstrings for these --- to be able to better communicate to the next contributors what these do.